app/src/main/java/com/github/HumanLearning2021/HumanLearningApp/model/UniqueDatabaseManagement.kt

```diff
@@ -0,0 +1,99 @@
+package com.github.HumanLearning2021.HumanLearningApp.model
+
+import android.content.Context
+import com.github.HumanLearning2021.HumanLearningApp.firestore.CachedFirestoreDatabaseManagement
+import com.github.HumanLearning2021.HumanLearningApp.firestore.FirestoreCategorizedPicture
+import com.github.HumanLearning2021.HumanLearningApp.firestore.FirestoreDatabaseManagement
+import com.github.HumanLearning2021.HumanLearningApp.firestore.FirestoreDatabaseService
+import com.github.HumanLearning2021.HumanLearningApp.offline.*
+import com.github.HumanLearning2021.HumanLearningApp.room.*
+
+class UniqueDatabaseManagement(private val context: Context) {
+
+    private val room = RoomOfflineDatabase.getDatabase(context)
+    private val databaseDao = room.databaseDao()
+    private val datasetDao = room.datasetDao()
+    private val categoryDao = room.categoryDao()
+
+    private val downloadedDatabases: MutableList<String> = mutableListOf()
+
+    init {
+        val dsNames = room.databaseDao().loadAll().map { db -> db.emptyHLDatabase.databaseName }
+        dsNames.forEach { dsName -> downloadedDatabases.add(dsName) }
+    }
+
+    fun getDownloadedDatabases(): List<String> = downloadedDatabases.toList()
+
+    suspend fun getDatabases(): List<String> = FirestoreDatabaseService.getDatabaseNames()
+
+    fun accessDatabase(databaseName: String): DatabaseManagement {
+        return if (downloadedDatabases.contains(databaseName)) {
+            OfflineDatabaseManagement(databaseName).initialize(context)
+        } else {
+            CachedFirestoreDatabaseManagement(databaseName).initialize(context)
+        }
+    }
+
+    fun accessDatabaseFromCloud(databaseName: String): CachedFirestoreDatabaseManagement {
```

Comment:
maybe
```suggestion
    fun accessCloudDatabase(databaseName: String): CachedFirestoreDatabaseManagement {
```
is a bit clearer?

Reply:
Indeed.